### PR TITLE
Report CLUSTER progress for append-optimized tables

### PR DESCRIPTION
**Pull request by l-wang**
Refer to Postgres documentation: https://www.postgresql.org/docs/current/progress-reporting.html#CLUSTER-PROGRESS-REPORTING

Variations for append-optimized tables include the following:

1. "seq scanning append-optimized" phase for AO tables maps to "seq scanning heap" phase for heap tables
2. "writing new append-optimized" phase for AO tables maps to "writing new heap" phase for heap tables
3. CLUSTER <ao_table> always performs scan-then-sort, so "index scanning XXX" phase is not applicable.
4. "heap_tuples_scanned" field for AO/AOCO tables only counts number of live tuples scanned.
5. "heap_blks_*" simply converts bytesize into equivalent heap blocks in size

PS: I will squash the last three commits. Reviewers can review either commit-by-commit or all together.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/isolation2/input/uao/cluster_progress.source [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+1<:
+
+-- cluster has finished, nothing should show up in the view.
+1U: select datname, relid::regclass, command, phase, cluster_index_relid::regclass, heap_tuples_scanned, heap_tuples_written, heap_blks_total, heap_blks_scanned, index_rebuild_count from pg_stat_progress_cluster;
+
+-- Cleanup
+SELECT gp_inject_fault_infinite('all', 'reset', dbid) FROM gp_segment_configuration;
+reset default_table_access_method;
+DROP TABLE cluster_progress_ao;
+DROP TABLE helper_table;
```
Comment (bmdoil) [inline src/test/isolation2/input/uao/cluster_progress.source:71]: nit: missing newline

**Review by bmdoil — APPROVED** [review]
LGTM, nice tests!

File: src/test/isolation2/input/uao/cluster_progress.source (hunk 1 above)
Comment (l-wang) [reply src/test/isolation2/input/uao/cluster_progress.source]: Fixed!

**Comment by l-wang** [comment]
Thank you @bmdoil for reviewing! Sorry about the force-push, I addressed your feedback, and also bumped the catalog version which I forgot earlier. Since changes are trivial (newline and catversion) and PR pipeline was happy, I'm merging now.

**Pull request merged**